app/src/main/java/com/example/fond/adapters/UserPostAdapter.java

```diff
@@ -66,11 +83,63 @@ public ViewHolder(View itemView) {
             tvUsername = itemView.findViewById(R.id.tvUsername);
             ivImage = itemView.findViewById(R.id.ivImage);
             tvCaption = itemView.findViewById(R.id.tvCaption);
+            chkFond = itemView.findViewById(R.id.chkFond);
+
+            chkFond.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+                @Override
+                public void onCheckedChanged(CompoundButton compoundButton, boolean isChecked) {
+                    if(isChecked){
+                        addFond(posts.get(getAdapterPosition()));
+                        fondedPosts.add(posts.get(getAdapterPosition()).getObjectId());
+                        Log.i(TAG, "Fonded posts" + fondedPosts.toString());
+                    } else {
+                        removeFond(posts.get(getAdapterPosition()));
+                        //remove from list
+                        fondedPosts.remove(posts.get(getAdapterPosition()).getObjectId());
+                    }
+                }
+            });
+
+        }
 
+        private void removeFond(UserPost userPost) {
+            ParseQuery<Fond> query = ParseQuery.getQuery("Fond");
+            query.whereEqualTo("postId", userPost);
+            query.whereEqualTo("userId", ParseUser.getCurrentUser());
+            query.getFirstInBackground(new GetCallback<Fond>() {
+                @Override
+                public void done(Fond object, ParseException e) {
+                    if (object != null){
+                        object.deleteInBackground();
+                    } else if (e != null) {
+                        Log.e(TAG, "Error removeFond: " + e);
+                    }
+                }
+            });
+        }
 
+        private void addFond(UserPost userPost) {
+            Fond fond = new Fond();
+
+            Log.i(TAG, "ParseUser" + ParseUser.getCurrentUser().toString());
+            Log.i(TAG, "UserPost" + userPost.toString());
+            fond.put("userId", ParseUser.getCurrentUser());
+            fond.put("postId", userPost);
+            fond.saveInBackground(new SaveCallback() {
+                @Override
+                public void done(ParseException e) {
+                    if (e != null){
```

Comment:
Line 131 might be part of the reason why an error log shows up when 'fonding' a post? I think it should be if `e == null`.

Reply:
(I went ahead and made the change in the interest of time b/c I'm pretty sure this was just a transcription error... lmk if I was wrong